5_problem-solving-patterns/3_chaeyoung.js

```diff
@@ -0,0 +1,58 @@
+// 3.
+// MULTIPLE POINTERS
+// Creating pointers or values that correspond to an index or position and move towards the beginning, end or middle based on a certain condition
+
+// Very efficient for solving problems with minimal space complexity as well
+
+// 3-1
+// AN EXAMPLE
+// Write a function called sumZero which accepts a sorted array of integers. The function should find the first pair where the sum is 0. Return an array that includes both values that sum to zero or undefined if a pair does not exist
+
+// sumZero([-3,-2,-1,0,1,2,3]) // [-3,3]
+// sumZero([-2,0,1,3]) // undefined
+// sumZero([1,2,3]) // undefined
+
+// SOLUTION
+// 목표 : 배열의 첫 번째 값과 마지막 값을 더하여 0이 되는 값을 찾는다.
+// 도무지 중첩 for문 없는 방법은 떠오르지 않아 강의 코드를 참고하였습니다.
+
+function sumZero(arr) {
+  let left = 0;
+  let right = arr.length - 1;
+  while (left < right) {
+    let sum = arr[left] + arr[right];
+    if (sum === 0) return [arr[left], arr[right]];
+    else if (sum > 0) right--;
+    else left++;
+  }
+}
```

Comment:
```suggestion
return undefined;
}
```
while문이 끝나면 조건 실패했다는 걸 명시하기 위해 return문을 달아줘도 갠찮을듯